cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/ObjectMappedTableProperties.java

```diff
@@ -40,22 +41,53 @@
   /**
    * The type of object in the table.
    */
-  public static final String TYPE = "type";
+  public static final String OBJECT_TYPE = "object.type";
 
   /**
    * The schema of objects in the table.
    */
-  public static final String SCHEMA = "schema";
+  public static final String OBJECT_SCHEMA = "object.schema";
+
+  /**
+   * The name of the key to use when exploring the table. Defaults to "key".
+   */
+  public static final String EXPLORE_KEY_NAME = "explore.key.name";
+
+  /**
+   * The type of the key when exploring the table. Defaults to bytes.
+   */
+  public static final String EXPLORE_KEY_TYPE = "explore.key.type";
```

Comment:
I'll take another pass at the javadocs.

For the second question, I think it is better to better to always include the row key because it is a more accurate representation of what is stored in the table, and because the HBaseStorageHandler always requires a row key. If we force it to be specified here it makes things more consistent once we add a way to create a Hive table that uses HBaseStorageHandler instead of our DatasetStorageHandler.